core/pkg/server/resume.go

```diff
@@ -12,12 +12,40 @@ import (
 	"github.com/wandb/wandb/core/pkg/service"
 )
 
+type Bucket = gql.RunResumeStatusModelProjectBucketRun
+
+type ResumeMode uint8
+
+const (
+	None ResumeMode = iota
+	Must
+	Allow
+	Never
+)
+
 type ResumeState struct {
-	ResumeMode       string // must, allow, never
+	resume           ResumeMode
 	FileStreamOffset fs.FileStreamOffsetMap
 	logger           *observability.CoreLogger
 }
 
+func resumeMode(mode string) ResumeMode {
+	switch mode {
+	case "must":
+		return Must
+	case "allow":
+		return Allow
+	case "never":
+		return Never
+	default:
+		return None
+	}
+}
+
+func NewResumeState(logger *observability.CoreLogger, mode string) *ResumeState {
```

Comment:
this should probably accept a ResumeMode rather than a string, and the caller can parse the string. Generally I'd do all parsing as early as possible

Reply:
fair, I was just changing it internally